pkg/server/server.go

```diff
@@ -272,3 +290,47 @@ func (s *Server) newEndpointsServer(catalog catalog.Catalog, svidRotator svid.Ro
 func (s *Server) caCertsPath() string {
 	return path.Join(s.config.DataDir, "certs.json")
 }
+
+func (s *Server) validateTrustDomain(ctx context.Context, ds datastore.DataStore) error {
+	trustDomain := s.config.TrustDomain.Host
+
+	fetchResponse, err := ds.ListRegistrationEntries(ctx, &datastore.ListRegistrationEntriesRequest{})
+	if err != nil {
+		s.config.Log.Error(err)
+		return errors.New("error trying to fetch entries")
+	}
+
+	for _, entry := range fetchResponse.Entries {
+		err = validateSpiffeId(entry.SpiffeId, trustDomain, invalidTrustDomainRegistrationEntry)
+		if err != nil {
+			return err
+		}
+	}
+
+	nodesResponse, err := ds.ListAttestedNodes(ctx, &datastore.ListAttestedNodesRequest{})
+	if err != nil {
+		s.config.Log.Error(err)
+		return errors.New("error trying to list attested nodes")
+	}
+
+	for _, node := range nodesResponse.Nodes {
+		err = validateSpiffeId(node.SpiffeId, trustDomain, invalidTrustDomainAttestedNode)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validateSpiffeId(spiffeId string, trustDomain string, errMsg string) error {
+	id, err := url.Parse(spiffeId)
+	if err != nil {
+		return fmt.Errorf("could not parse SPIFFE ID: %v", err)
```

Comment:
it would be good if we could include more information here... that it is a registration entry that is bad,  and perhaps the entry ID so the user can inspect it/delete it